core/src/main/java/hivemall/smile/tools/MatrixEventStream.java

```diff
@@ -0,0 +1,79 @@
+package hivemall.smile.tools;
+
+import java.util.Iterator;
+import java.util.List;
+
+import hivemall.math.matrix.Matrix;
+import hivemall.smile.data.Attribute;
+import hivemall.smile.data.Attribute.AttributeType;
+import opennlp.model.AbstractEventStream;
+import opennlp.model.Event;
+
+public class MatrixEventStream extends AbstractEventStream {
+	  Matrix ds;
+	  int[] y;
+	  Attribute[] attrs;
+	  int rowNum;
+	  Event next;
+
+	  public MatrixEventStream (Matrix ds, int[] y, Attribute[] attrs) {
+	    this.ds = ds;
+	    this.y = y;
+	    this.attrs = attrs;
+	    rowNum = 0;
+	    
+	    if (rowNum < ds.numRows()){
+	      next = createEvent(ds.getRow(rowNum), y[rowNum]);
+	    };
+	  }
+	  	  
+	  /**
+	   * Returns the next Event object held in this EventStream.  Each call to nextEvent advances the EventStream.
+	   *
+	   * @return the Event object which is next in this EventStream
+	   */
+	  public Event next () {
+	    while (next == null && (rowNum < ds.numRows()))
```

Comment:
`{}` is needed to [Google coding style](https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style) that Hivemall follows.